### ISSUE TITLE
Crash when trying to get the type of an alias that isn't resolved correctly

### ISSUE THREAD

**Issue by leizongmin** [Microsoft/TypeScript#15094]
**TypeScript Version:**  2.2.2

**Code**

```tsx
import React from 'reactxp';
import {
  Component,
} from 'reactxp';

class HelloReactXP extends Component {
  render() {
    return (
      <View>
        <Text>
          Hello, ReactXP
        </Text>
      </View>
    );
  }
}
```

**Expected behavior:** Success

**Actual behavior:** Throws following error and exited:

```
/usr/local/lib/node_modules/typescript/lib/tsc.js:51296
                throw e;
                ^

TypeError: Cannot read property 'kind' of undefined
    at getRootDeclaration (/usr/local/lib/node_modules/typescript/lib/tsc.js:6638:20)
    at Object.isCatchClauseVariableDeclarationOrBindingElement (/usr/local/lib/node_modules/
typescript/lib/tsc.js:5228:20)
    at getTypeOfVariableOrParameterOrProperty (/usr/local/lib/node_modules/typescript/lib/ts
c.js:22788:24)
    at getTypeOfSymbol (/usr/local/lib/node_modules/typescript/lib/tsc.js:22956:24)
    at getTypeOfAlias (/usr/local/lib/node_modules/typescript/lib/tsc.js:22922:23)
    at getTypeOfSymbol (/usr/local/lib/node_modules/typescript/lib/tsc.js:22968:24)
    at checkIdentifier (/usr/local/lib/node_modules/typescript/lib/tsc.js:28674:24)
    at checkExpressionWorker (/usr/local/lib/node_modules/typescript/lib/tsc.js:32221:28)
    at checkExpression (/usr/local/lib/node_modules/typescript/lib/tsc.js:32205:42)
    at getBaseConstructorTypeOfClass (/usr/local/lib/node_modules/typescript/lib/tsc.js:2307
9:43)
```

**Comment by sandersn** [comment]
Here's how I reproduced this bug:

create `test.tsx` and `tsconfig.json` in a new directory. In `tsconfig.json`, put

```json
{
    "compilerOptions": {
        "jsx": "preserve"
    },
    "files": [
        "test.tsx"
        ]
}
```

Then run

```sh
$ npm install reactxp
$ tsc
```

**Comment by sandersn** [comment]
Still repros on master

**Comment by sandersn** [comment]
Here's a smaller repro. In test.ts (not test.tsx), put:

```ts
import { Component } from 'reactxp';
class HelloReactXP extends Component { }
```
In `tsconfig.json`, put

```json
{
    "files": [ "test.ts" ]
}
```

Then run

```sh
$ npm install reactxp
$ tsc
```

The crash is not related to TSX, but to the `reactxp` package. The crash happens when trying to get the type of `Component`, which doesn't resolve for some reason, returning `unknownSymbol`, which then crashes `getTypeOfSymbol` because it doesn't expect to see `unknownSymbol`, which has no `valueDeclaration` (or any declarations at all).

This seems like two, possibly three bugs.

1. `Component` should resolve since `reactxp` has its own typings.
2. `getTypeOfAlias` should probably handle `unknownSymbol` itself in a special case.
3. Or `getTypeOfSymbol` shouldn't crash when a symbol doesn't have any declarations.

(3) may be intentional but (1) and (2) both seem incorrect. I'm going to go find out why `Component` didn't resolve. Maybe `reactxp` has its typings set up incorrectly.

**Comment by sandersn** [comment]
The reason for (1) is that `reactxp` doesn't have `react` as a dependency, so when `reactxp` does `import React = require('react'); export import Component = React.Component;`, `React.Component` can't be resolved. As soon as I install `react` and `@types/react`, the crash goes away. That's a workaround for now.

**Comment by sandersn** [comment]
Here's the final repro, which doesn't depend on module resolution either:

```ts
// @Filename: second.d.ts
export import Component = CompletelyMissing;
// @Filename: first.d.ts
import * as Second from './second';
export = Second;
// @Filename: crash.ts
import { Component } from './first';
class C extends Component { }
```

**Comment by sandersn** [comment]
Fix is up at #16194
